e2e_tests/provider_cfg/cryptoauthlib/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM ubuntu:18.04
```

Comment:
Thanks for adding the testing infrastructure for this provider as well!
I guess you will add a new CI workflow for this provider in the `.github/workflows/ci.yml` file later on? I don't think it's needed to do now, since most test will fail and the building part is handled by the "all-providers" test.

Reply:
Yes, at the moment there is just a very basic structure. We will work on enabling tests later on.

Reply:
Unfortunately my comment was very wrong. If the CI tests are to be enabled for this particular provider, there must be underlying ATECC hardware available, which is not an issue for our team, but I can imaging it is a challenge on a public github. 
Is it possible on github at all?

Reply:
So I take it there's no way to have a simulator, or even a mock library that returns preconfigured responses? We could run a different set of tests for this particular provider in that case

Reply:
Yes that would be a problem. Currently all of our tests are using virtualized environment: virtual TPM server and mock PKCS11 library for example. I was hoping that there would be something similar for either a virtualised ATECCx08 device that the CryptoAuth lib can communicate to or a mock CryptoAuth lib implementation that would work standalone for testing.
A quick Google did not show me anything useful :/

If it's not possible, maybe for now we can limit ourselves to local end-to-end testing (but still do build and lint tests on the CI). Of course this is not a viable option in the long run amd we will need to think on how to support this.

Reply:
Thank you for quick response.
To be honest - the cryptochip is not being accessed directly but through rust-cryptoauthlib, and there might be implemented an additional "device" that uses a software implementation (based on e.g. mbed-tls). But this is an additional, not planned effort and its purpose would be limited to testing only.

Reply:
Sure, that makes sense - we could merge this now and raise an issue to look into testing later, even if it's just for sanity checking. 

I think it makes sense to simply not include that provider in E2E tests, but even just unit tests - if you can manage to get some to run, maybe with mocking - would help.